src/core/PlayArea.ttslua

```diff
@@ -85,130 +67,54 @@ function getClueCount(object, isFaceDown, playerCount)
-        token.setCustomObject(custom)
-        token.scale {0.18, 1, 0.18}
-        return token
-      end
-  end
-
-  if number == '3' or number == '4' then
-    obj_parameters.type = 'Custom_Tile'
-    custom.type = 2
-    local token = spawnObject(obj_parameters)
-    if number == '3' then
-      custom.image = TOKEN_PLAYER_ONE.clue
-      custom.image_bottom = TOKEN_PLAYER_ONE.doom
-      token.setCustomObject(custom)
-      token.scale {0.25, 1, 0.25}
-      token.use_snap_points=false
-      return token
-    end
-
-    if number == '4' then
-      custom.image = TOKEN_PLAYER_ONE.doom
-      custom.image_bottom = TOKEN_PLAYER_ONE.clue
-      token.setCustomObject(custom)
-      token.scale {0.25, 1, 0.25}
-      token.use_snap_points=false
-      return token
-    end
-  end
-
-end
-
-
-function spawnCluesAtLocation(clueCount, collision_info)
-  local object = collision_info.collision_object
-  if isAlreadySpawned(object) then
+function spawnCluesAtLocation(clueCount, object)
+  if SPAWNED_LOCATION_GUIDS[object.getGUID()] ~= nil then
     error('tried to spawn clue for already spawned location:' .. object.getName())
   end
 
-  local obj_parameters = {}
-  obj_parameters.type = 'Custom_Token'
-  obj_parameters.position = {
-    object.getPosition()[1],
-    object.getPosition()[2] + 1,
-    object.getPosition()[3]
-  }
-
   log('spawning clues for ' .. object.getName() .. '_' .. object.getGUID())
-  local playerCount = COUNTER.getVar('val')
-  log('player count is ' .. playerCount .. ', clue count is ' .. clueCount)
+  log('player count is ' .. COUNTER.getVar('val') .. ', clue count is ' .. clueCount)
+
   -- mark this location as spawned, can't happen again
-  markSpawned(object)
-  i = 0
-  while i < clueCount do
-    if i < 4 then
-      obj_parameters.position = {
-        collision_info.collision_object.getPosition()[1] + 0.3,
-        collision_info.collision_object.getPosition()[2] + 0.2,
-        collision_info.collision_object.getPosition()[3] - 0.8 + (0.55 * i)
-      }
-    elseif i < 8 then
-      obj_parameters.position = {
-        collision_info.collision_object.getPosition()[1] + 0.85,
-        collision_info.collision_object.getPosition()[2] + 0.2,
-        collision_info.collision_object.getPosition()[3] - 3 + (0.55 * i)
-      }
-    else
-      obj_parameters.position = {
-        collision_info.collision_object.getPosition()[1] + 0.575,
-        collision_info.collision_object.getPosition()[2] + 0.4,
-        collision_info.collision_object.getPosition()[3] - 5.2 + (0.55 * i)}
-    end
-    spawnToken(obj_parameters.position, '3')
-    i = i + 1
+  SPAWNED_LOCATION_GUIDS[object.getGUID()] = 1
+
+  -- spawn clues (starting top right, moving to the next row after 4 clues)
+  local pos = object.getPosition()
+  for i = 1, clueCount do
+    local row = math.floor(1 + (i - 1) / 4)
+    local column = (i - 1) % 4
+    spawnClues({ pos[1] + 1.5 - 0.55 * row, pos[2], pos[3] - 0.825 + 0.55 * column })
```

Comment:
pos should have .x, .y, .z values, much clearer to use those instead of the number index